app/models/user.rb

```diff
@@ -1,9 +1,13 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
   has_many :posts, dependent: :destroy, foreign_key: 'author_id'
   has_many :comments, dependent: :destroy, foreign_key: 'users_id'
   has_many :likes, dependent: :destroy, foreign_key: 'users_id'
   validates :name, presence: true
-  validates :posts_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }, presence: true
+  validates :posts_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
```

Comment:
- [ ] Great job team on the implementation of user authentication and registration :clap: However your validation for `posts_counter` is currently preventing the registration of new users (see image). A better approach to ensure that the `posts_counter` column values are numeric and greater than 0 would be to remove the validation from the model and set a default value for the `posts_counter` column directly in the migration. You can do this like so:

```ruby
# Generate a new migration file with rails g migration migration_name
def change
  migration_name :users, :posts_counter, :integer, default: 0
end
```
![Screenshot from 2023-10-06 09-21-05](https://github.com/yemidada/blog-app/assets/91142432/5fd3e94a-a56c-49ff-a207-8d0556561b58)